File: 01-inradius-of-a-right-triangle/main.py
#! /usr/bin/python

def main():
    leg_a = float(input("Введіть катет (a): "))            # Катет a
    hypotenuse = float(input("Введіть гіпотенузу (c): "))  # Гіпотенуза

    leg_b = (hypotenuse**2 - leg_a**2) ** 0.5  # Катет b за теоремою піфагора
    radius = (leg_a + leg_b - hypotenuse) / 2  # Радіус вписаного кола
    print("Радіус вписаного кола (r):", radius)


if __name__ == "__main__":
    main()

File: 01-inradius-of-a-right-triangle/main.cpp
#include <stdio.h>
#include <cmath>


int main() {
    float leg_a, leg_b,  // Катети (a та b)
          hypotenuse,    // Гіпетенуза (c)
          radius;        // Радіус вписаного кола

    printf("Введіть катет (a): "); scanf("%f", &leg_a);
    printf("Введіть гіпотенузу (c): "); scanf("%f", &hypotenuse);

    leg_b = sqrt(pow(hypotenuse, 2) - pow(leg_a, 2));  // Катет b за теоремою піфагора
    radius = (leg_a + leg_b - hypotenuse) / 2;         // Радіус вписаного кола

    printf("Радіус вписаного кола (r): %g\n", radius);

    return 0;
}
